kyaml/yaml/rnode.go

```diff
@@ -565,6 +565,10 @@ func (rn *RNode) SetMapField(value *RNode, path ...string) error {
 	)
 }
 
+func (rn *RNode) AssocMapEntry(key, value *RNode) error {
```

Comment:
nit: ReplaceMapEntry? 

I'm not sure we need this though. Generally kyaml consumers probably don't care to preserve the entire RNode properties like we do in the super low-level case in this PR. So I'd be inclined to just move the code below inline (where we know we have a map, which as a sidenote, we would need to check if this method stays).

Reply:
nit: `Replace` for me indicates it should fail key doesn't exist, `Assoc[iate]` does not. Maybe `Put` or `Upsert`? 

Since this mess up with other tests, especially on JSON serialization, I think it's quite important to maintain the properties